### PR TITLE
[CUMULUS-2497] Add support to get granule temporal info from CMR files in ISO format

### PR DESCRIPTION
**Pull request by botanical**
**Summary:** Summary of changes

Addresses [CUMULUS-2497: Add support to get temporal info from CMR files in ISO format](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2497)

## Changes

* These changes are for retrieving `granule` temporal information from a CMR ISO SMAP XML file.
* ISO MENDS files exist but are this work is only focusing on SMAP as per user request.

Because the keys contain a `:`, I access the values of the keys using `[]` rather than dot notation. For context, ISO 19115 is a tool to define metadata for geographic information. It provides info about identification, quality, extent, spatial and temporal schemas, and distribution of geographic data.

- `MD_DataIdentification:extent` contains info on the temporal extent of the dataset with the path being 
```
/gmd:DS_Series/gmd:seriesMetadata/gmi:MI_Metadata/gmd:identificationInfo/
gmd:MD_DataIdentification/gmd:extent/gmd:EX_Extent/gmd:temporalElement/
gmd:EX_TemporalExtent/gmd:extent/gml:TimePeriod/gml:beginPosition
```
More info can be found [here](https://wiki.earthdata.nasa.gov/display/CMR/Temporal+Extent)
- `LE ProcessStep` contains information on when the process when completed. The info can be retrieved from the following path:
```
/gmi:MI_Metadata/gmd:dataQualityInfo/gmd:DQ_DataQuality/gmd:lineage/gmd:LI_Lineage/gmd:processStep/gmi:LE_ProcessStep/gmd:dateTime/gco:DateTime
```
More info on `ProductionDateTime` can be found [here](https://wiki.earthdata.nasa.gov/display/CMR/Production+Date+Time#ProductionDateTime-DialectMappings)
- `lastUpdateTime` can be found with the following path:
```
/gmd:DS_Series/gmd:composedOf/gmd:DS_DataSet/gmd:has/gmi:MI_Metadata/gmd:identificationInfo/gmd:MD_DataIdentification/gmd:citation/gmd:CI_Citation/gmd:date/gmd:CI_Date/gmd:date/gco:DateTime {the update time}

where

/gmd:DS_Series/gmd:composedOf/gmd:DS_DataSet/gmd:has/gmi:MI_Metadata/gmd:identificationInfo/gmd:MD_DataIdentification/gmd:citation/gmd:CI_Citation/gmd:title/gco:CharacterString = UpdateTime
```
More information on this can be found [here](https://wiki.earthdata.nasa.gov/display/CMR/Provider+Dates)
## PR Checklist

- [x] Update CHANGELOG
- [x] Unit tests
- [ ] Adhoc testing
- [ ] Integration tests

### CODE REVIEW

File: packages/cmrjs/tests/cmr-utils/test-cmr-utils.js [hunk 1]
```diff
@@ -110,56 +110,35 @@ test.after.always(async (t) => {
...
 
-test('isCMRFile returns truthy if fileobject has valid xml name', (t) => {
+test('isCMRFile returns truthy if fileobject has valid xml filename', (t) => {
   const fileObj = {
-    name: 'validfile.cmr.xml',
-  };
-  t.truthy(isCMRFile(fileObj));
-});
-
-test('isCMRFile returns falsy if fileobject does not valid xml name', (t) => {
```
Comment (botanical) [inline packages/cmrjs/tests/cmr-utils/test-cmr-utils.js:120]: There seemed to be a couple duplicate unit tests so I've removed them

File: packages/cmrjs/src/cmr-utils.js [hunk 2]
```diff
@@ -991,6 +993,27 @@ async function getGranuleTemporalInfo(granule) {
...
 
   const cmrFilename = cmrFile[0].filename;
+
+  if (isISOFile(cmrFilename)) {
+    const metadata = await metadataObjectFromCMRXMLFile(cmrFilename);
+    const metadataMI = metadata['gmd:DS_Series']['gmd:composedOf']['gmd:DS_DataSet']['gmd:has']['gmi:MI_Metadata'];
+
+    // Get beginning and ending date time from beginPosition and endPosition
+    const identificationInfo = metadataMI['gmd:identificationInfo'];
+    const dataIdentification = identificationInfo.find((dataIdObject) => Object.keys(dataIdObject).filter((key) => Object.keys(dataIdObject[key]).includes('gmd:extent')));
```
Comment (markdboyd) [inline packages/cmrjs/src/cmr-utils.js:1003]: You're right: this is is some gnarly XML parsing code, but it looks good to me 👍

**Comment by laurenfrederick** [comment]
@markdboyd NSIDC currently has their own tasks to deal with ISO metadata for posting to CMR, so that is out of scope for this ticket. This ticket is to get the correct info over to Cloud Metrics.

**Comment by schwabm** [comment]
Hi, 

 The code that Jennifer wrote, if it has access to our ISO XML metadata records which are in CMR should find all four of the timestamps her code is looking for using the XPaths she has coded.  
 So, as long as the EMS utility runs this code, and you are not expecting us to have run it during our data ingest or metadata export to CMR workflows it should work. To be clear, we do not use the post_to_cmr cumulus core task to export metadata records to  the CMR.  We have our own export utility which does that.

Mark

**Review by markdboyd — APPROVED** [review]

File: packages/cmrjs/src/cmr-utils.js [hunk 3]
```diff
@@ -64,7 +64,11 @@ function getFileDescription(file, urlType = 'distribution') {
 
 const isECHO10File = (filename) => filename.endsWith('cmr.xml');
 const isUMMGFile = (filename) => filename.endsWith('cmr.json');
-const isCMRFilename = (filename) => isECHO10File(filename) || isUMMGFile(filename);
+const isISOFile = (filename) => filename.endsWith('cmr_iso.xml');
+// eslint-disable-next-line max-len
```
Comment (markdboyd) [inline packages/cmrjs/src/cmr-utils.js:68]: Do you still need this?

Comment (botanical) [reply packages/cmrjs/src/cmr-utils.js]: definitely didn't mean to leave that in there~ thanks for letting me know, it's been removed [`b9fe488` (#2226)](https://github.com/nasa/cumulus/pull/2226/commits/b9fe4886adc6c8e00996252e9e65e9d2529134c5)

**Pull request merged**